Shop.BLL/Services/AuthService.cs

```diff
@@ -0,0 +1,69 @@
+using System.Text;
+using AutoMapper;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Logging;
+using Shop.BLL.Common.DataTransferObjects.Users;
+using Shop.BLL.Exceptions.InternalExceptions;
+using Shop.BLL.Interfaces;
+using Shop.DAL.Models;
+
+namespace Shop.BLL.Services
+{
+    public class AuthService : IAuthService
+    {
+        private readonly IMapper _mapper;
+        private readonly UserManager<User> _userManager;
+        private readonly ILogger<AuthService> _logger;
+
+        public AuthService(IMapper mapper, UserManager<User> userManager,
+            ILogger<AuthService> logger)
+        {
+            _mapper = mapper;
+            _userManager = userManager;
+            _logger = logger;
+        }
+        public async Task<bool> AuthorizeUserAsync(UserRequestAuthorizationDto userRequestAuthorizationDto)
+        {
+            var user = await _userManager
+                .FindByEmailAsync(userRequestAuthorizationDto.Email);
+
```

Comment:
it is necessary to check whether such a user exists and if not, then throw the appropriate exception like if(user is null){...}

Reply:
If notfound exception is thrown, global exception handler will capture exception and return 404, which let client know, that user with that name does not exist which is vulnerability. I suppose its better just return false to controller and then return Unauthorized